src/one/util/ideaplugin/screenshoter/CopyImageAction.java

```diff
@@ -1,155 +1,66 @@
-        AffineTransform newTransform = new AffineTransform(currentTransform);
-        newTransform.scale(scale, scale);
-        // To flush glyph cache
-        paint(contentComponent, newTransform, 1, 1);
-        Rectangle2D r = new Rectangle2D.Double();
-
-        for (int i = start; i <= end; i++) {
-            if (options.myChopIndentation &&
-                    EMPTY_SUFFIX.matcher(text.substring(0, Math.min(i - start + 1, text.length()))).find()) {
-                continue;
-            }
-            VisualPosition pos = editor.offsetToVisualPosition(i);
-            Point2D point = getPoint(editor, pos);
-            includePoint(r, point);
-            includePoint(r, new Point2D.Double(point.getX(), point.getY()+editor.getLineHeight()));
-        }
+        Editor editor = geEditor(event);
+        if (editor == null) return;
 
-        newTransform.translate(-r.getX(), -r.getY());
-        BufferedImage image = paint(contentComponent, newTransform,
-                (int) (r.getWidth() * scale), (int) (r.getHeight() * scale));
-
-        selectionModel.setSelection(start, end);
-        caretModel.moveToOffset(offset);
+        Image image = new ImageBuilder(editor).createImage();
 
         TransferableImage transferableImage = new TransferableImage(image);
         Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
-        clipboard.setContents(transferableImage, (clipboard1, contents) -> { });
-    }
-
-    private static void includePoint(Rectangle2D r, Point2D p) {
-        if(r.isEmpty()) {
-            r.setFrame(p, new Dimension(1, 1));
-        } else {
-            r.add(p);
-        }
-    }
-
-    @NotNull
-    private static BufferedImage paint(JComponent contentComponent, AffineTransform at, int width, int height) {
-        BufferedImage img = null;
-        if(utilCreateImage != null) {
-            try {
-                img = (BufferedImage) utilCreateImage.invoke(null, contentComponent, width, height, BufferedImage
-                        .TYPE_INT_RGB);
-            } catch (IllegalAccessException | InvocationTargetException ignored) {
-            }
-        }
-        if(img == null) {
-            //noinspection UndesirableClassUsage
-            img = new BufferedImage(width, height, BufferedImage.TYPE_INT_RGB);
-        }
-        Graphics2D graphics = (Graphics2D) img.getGraphics();
-        graphics.setTransform(at);
-        contentComponent.paint(graphics);
-        return img;
-    }
-
-    private Point2D getPoint(Editor editor, VisualPosition pos) {
-        return editor.visualPositionToXY(pos);
+        clipboard.setContents(transferableImage, (clipboard1, contents) -> {
+        });
     }
-
+    
     @Override
     public void update(AnActionEvent event) {
         Presentation presentation = event.getPresentation();
+        presentation.setEnabled(geEditor(event) != null);
+    }
+
+    static Editor geEditor(@NotNull AnActionEvent event) {
```

Comment:
geEditor?

Reply:
sorry, I didn't get the point. Could you please explain?

Reply:
Should be "getEditor". Nevermind, will fix this by myself.